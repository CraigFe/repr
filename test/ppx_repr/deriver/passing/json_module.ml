(* Ensure that the [Json] module in [Repr.Type] doesn't shadow references to
   types contained in a different [Json] module.

   Regression test for https://github.com/mirage/irmin/issues/923. *)

module Json = struct
  type t = string

  let t = Repr.Type.string
end

type foo = { contents : Json.t } [@@deriving repr]

let (_ : foo Repr.Type.t) = foo_t
