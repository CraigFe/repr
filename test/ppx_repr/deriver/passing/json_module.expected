module Json = struct type t = string
                     let t = Repr.Type.string end
type foo = {
  contents: Json.t }[@@deriving repr]
include
  struct
    let foo_t =
      Repr.Type.sealr
        (Repr.Type.(|+)
           (Repr.Type.record "foo" (fun contents -> { contents }))
           (Repr.Type.field "contents" Json.t (fun t -> t.contents)))
  end[@@ocaml.doc "@inline"][@@merlin.hide ]
let (_ : foo Repr.Type.t) = foo_t
